core/api/src/servers/middlewares/scope.ts

```diff
@@ -0,0 +1,79 @@
+import { GraphQLResolveInfo, GraphQLFieldResolver } from "graphql"
+
+import ScopesOauth2 from "@/domain/authorization"
+import { AuthorizationError } from "@/domain/errors"
+import { mapError } from "@/graphql/error-map"
+import { mutationFields } from "@/graphql/public"
+
+const readTransactionsAuthorize = async (
+  resolve: GraphQLFieldResolver<unknown, GraphQLPublicContextAuth>,
+  parent: unknown,
+  args: unknown,
+  context: GraphQLPublicContextAuth,
+  info: GraphQLResolveInfo,
+) => {
+  const scope = context.scope
+  const appId = context.appId
+
+  // not a delegated token
+  if (appId === undefined || appId === "") {
+    return resolve(parent, args, context, info)
```

Comment:
does resolve mean that we have been authorised? If so this seem likes a potential issue - I think you mean `&&` here not `||` - right?

Reply:
so, currently, if scope is undefined or appId is undefined, it means this is a kratos authenticated request, not a hydra authenticated request.

if this is a kratos request, it means it's a "primary"/first party user request. then we don't need to look into this middleware because it's solely for delegation request

the way to skip this middleware is to resolve it right away and not continue the function.